Sources/MatrixClient/API/Events/Messages/Messages.swift

```diff
@@ -0,0 +1,433 @@
+//
+//  File.swift
+//
+//
+//  Created by Finn Behrens on 13.03.22.
+//
+
+import Foundation
+
+public struct MatrixMessageRelatesTo: Codable {
+    public var inReplyTo: InReplyTo?
+
+    enum CodingKeys: String, CodingKey {
+        case inReplyTo = "m.in_reply_to"
+    }
+
+    public struct InReplyTo: Codable {
+        public var eventID: String
```

Comment:
Not sure I understand. `String?` will be nil if not present. so this would also be nil, as long as nil is a option.

Reply:
Sorry I was thinking if `eventID` isn't present in the JSON, then decoding will fail here though as you can't assign `nil` to `String`. And I was wondering if the result of that would be to throw all the way back to `JSONDecoder.decode` (and make the sync fail) or if it would just set `InReplyTo` to nil. I can't remember off the top of my head the behaviour here.

Reply:
good question. just let us make it optional, and be sure that it does not break.